docs/backup-restore-integration.md

```diff
@@ -0,0 +1,349 @@
+...
+```
+
+- ("", "PersistentVolumeClaim", "ns1", "pvc1")
+
+#### spec.networks[\*].multus
+
+```yaml
+...
+spec:
+  networks:
+  - name: ovs-net
+    multus:
+      networkName: ovs-vlan-100
+...
+```
+
+- ("k8s.cni.cncf.io", "NetworkAttachmentDefinition", "ns1", "ovs-vlan-100")
+
+#### spec.accessCredentials[\*].sshPublicKey
+
+```yaml
+...
+spec:
+  accessCredentials:
+  - sshPublicKey:
+      source:
+        secret:
+          secretName: my-pub-key
+...
+```
+
+- ("", "Secret", "ns1", "my-pub-key")
+
+#### spec.accessCredentials[\*].userPassword
+
+```yaml
+...
+spec:
+  accessCredentials:
+  - userPassword:
+      source:
+        secret:
+          secretName: my-user-password
+...
+```
+
+- ("", "Secret", "ns1", "my-user-password")
+
+### VirtualMachineInstanceReplicaSet Object Graph
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstanceReplicaSet
+metadata:
+  name: vmirs1
+  namespace: ns1
+...
+```
+
+- ("kubevirt.io", "VirtualMachineInstanceReplicaSet", "ns1", "vmirs1")
+- ("kubevirt.io", "VirtualMachineInstance", "ns1", "vmirs1XXXX1") \*
+- ("kubevirt.io", "VirtualMachineInstance", "ns1", "vmirs1XXXX2") \*
+
+\* there are usually multiple VirtualMachineInstances corresponding to a VirtualMachineInstanceReplicaSet.  The backup process can look up the name of this pod by using the `kubevirt.io/vmReplicaSe=<name of VirtualMachineInstanceReplicaSet>` label selector.
+
+#### spec.template
+
+See [VirtualMachineInstance](#virtualmachineinstance-object-graph)
+
+## Backup Actions
+
+### Guest filesystem freeze/thaw hooks
+
+See [this guide](https://github.com/kubevirt/kubevirt/blob/main/docs/freeze.md) for how to execute the freeze/thaw hooks for each VirtualMachineInstance encountered in the object graph.
+
+## Restore Actions
+
+### VirtualMachine Restore
+
+If restoring to a different cluster, and mac address or bios serial number are explicitly set, you should make sure there will be no collisions.  These values are set in:
+
+```
+/spec/template/spec/domain/devices/interfaces/<index>/macAddress
+/spec/template/spec/domain/firmware/serial
+```
+
+### VirtualMachineInstance Restore
+
+If a VirtualMachineInstance is **owned by a VirtualMachine**, it should **not be restored**.  The KubeVirt controller will recreate the resource based on the VirtualMachine definition.  Otherwise, VirtualMachineInstance definitions may be backed up/restored with the same precautions as VirtualMachine for mac address/bios.
+
+```
+/spec/domain/devices/interfaces/<index>/macAddress
+/spec/domain/firmware/serial
+```
+
+### virt-launcher Pod Restore
+
+A Pod with “virt-launcher-” prefix that is ***owned by a VirtualMachineInstance** should not be restored.
+
+### DataVolume Restore
+
+DataVolumes in **Succeeded phase** (status.phase) should have the following **annotation added** at restore time.  Otherwise, the associated PersistentVolumeClaim may get corrupted.  DataVolumes in any phase other than Succeeded do not need to be annotated.
```

Comment:
2 questions:
1. Shouldn't the annotation on the dv be on the backup phase? we said that the status of the object may not be restored?
2. dont we need to mark the PVC s that are in progress with some annotation to keep from restoring it like we do in velero plugin?

Reply:
More I think about it, more I believe it would be bad to not restore resource status.  Will check what velero/others do.

I don't think point 2 is an issue if status is restored

Reply:
In order to not make the restore fail in case of PVC inprogress we need to prevent the restore to restore it because in that case if the dv will create a new PVC before the backed up PVC is restored then the restore will fail cause the PVC already exist.